(just for myself for later)

```
======================================================================
FAIL: test_case_reindexer (testapps.test_pillowtop.tests.test_reindexer.PillowtopReindexerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/testapps/test_pillowtop/tests/test_reindexer.py", line 36, in test_case_reindexer
    self.assertEqual(1, results.total)
AssertionError: 1 != 2
----------------------------------------------------------------------
```

--- a/corehq/tests/app_test_db_dependencies.yml
+++ b/corehq/tests/app_test_db_dependencies.yml
@@ -16,6 +16,9 @@ testapps.test_pillowtop:
   - corehq.apps.hqadmin
   - corehq.apps.users
   - corehq.couchapps
+  - corehq.form_processor
+  - corehq.sql_accessors
+  - corehq.sql_proxy_accessors
   - couchforms
   - pillowtop
 casexml.apps.phone:

--- a/corehq/util/elastic.py
+++ b/corehq/util/elastic.py
@@ -1,4 +1,5 @@
 from django.conf import settings
+from elasticsearch import NotFoundError
 
 from corehq.util.test_utils import unit_testing_only
 
@@ -8,6 +9,14 @@ TEST_ES_PREFIX = 'test_'
 def es_index(index):
     prefix = '' if not settings.UNIT_TESTING else TEST_ES_PREFIX
     return "{}{}".format(prefix, index)
+
+
+@unit_testing_only
+def ensure_index_deleted(es_index):
+    try:
+        delete_es_index(es_index)
+    except NotFoundError:
+        pass
 
 
 @unit_testing_only

--- a/testapps/test_pillowtop/tests/test_reindexer.py
+++ b/testapps/test_pillowtop/tests/test_reindexer.py
@@ -5,7 +5,7 @@ from casexml.apps.case.models import CommCareCase
 from casexml.apps.case.signals import case_post_save
 from corehq.apps.es import UserES, CaseES, FormES, ESQuery
 from corehq.apps.users.models import CommCareUser
-from corehq.form_processor.tests.utils import TestFormMetadata
+from corehq.form_processor.tests.utils import TestFormMetadata, FormProcessorTestUtils
 from corehq.pillows.mappings.user_mapping import USER_INDEX
 from corehq.util.context_managers import drop_connected_signals
 from corehq.util.elastic import delete_es_index
@@ -26,8 +26,10 @@ class PillowtopReindexerTest(TestCase):
         self.assertEqual(self.domain, user_doc['domain'])
         self.assertEqual(username, user_doc['username'])
         self.assertEqual('CommCareUser', user_doc['doc_type'])
+        delete_es_index(USER_INDEX)
 
     def test_case_reindexer(self):
+        FormProcessorTestUtils.delete_all_cases()
         case_name = 'reindexer-test-case-{}'.format(uuid.uuid4().hex)
         with drop_connected_signals(case_post_save):
             CommCareCase(domain=self.domain, name=case_name).save()
@@ -72,3 +74,4 @@ class PillowtopReindexerTest(TestCase):
         self.assertEqual(user_id, user_doc['_id'])
         self.assertEqual('UnknownUser', user_doc['doc_type'])
         form.delete()
+        delete_es_index(USER_INDEX)

--- a/testapps/test_pillowtop/tests/test_reindexer.py
+++ b/testapps/test_pillowtop/tests/test_reindexer.py
@@ -42,6 +42,7 @@ class PillowtopReindexerTest(TestCase):
         self.assertEqual('CommCareCase', case_doc['doc_type'])
 
     def test_xform_reindexers(self):
+        FormProcessorTestUtils.delete_all_xforms()
         metadata = TestFormMetadata(domain=self.domain)
         form = make_es_ready_form(metadata).wrapped_form
         form.save()
@@ -55,6 +56,7 @@ class PillowtopReindexerTest(TestCase):
         form.delete()
 
     def test_unknown_user_reindexer(self):
+        FormProcessorTestUtils.delete_all_xforms()
         user_id = 'test-unknown-user'
         metadata = TestFormMetadata(domain=self.domain, user_id='test-unknown-user')
         form = make_es_ready_form(metadata).wrapped_form

--- a/testapps/test_pillowtop/tests/test_reindexer.py
+++ b/testapps/test_pillowtop/tests/test_reindexer.py
@@ -8,7 +8,7 @@ from corehq.apps.users.models import CommCareUser
 from corehq.form_processor.tests.utils import TestFormMetadata, FormProcessorTestUtils
 from corehq.pillows.mappings.user_mapping import USER_INDEX
 from corehq.util.context_managers import drop_connected_signals
-from corehq.util.elastic import delete_es_index
+from corehq.util.elastic import delete_es_index, ensure_index_deleted
 from corehq.util.test_utils import make_es_ready_form
 
 
@@ -59,7 +59,7 @@ class PillowtopReindexerTest(TestCase):
         metadata = TestFormMetadata(domain=self.domain, user_id='test-unknown-user')
         form = make_es_ready_form(metadata).wrapped_form
         form.save()
-        delete_es_index(USER_INDEX)
+        ensure_index_deleted(USER_INDEX)
         call_command('ptop_fast_reindex_unknownusers', noinput=True, bulk=True)
         # the default query doesn't include unknown users so should have no results
         self.assertEqual(0, UserES().run().total)

